src/storage/mvcc/txn.rs

```diff
@@ -163,10 +174,18 @@ impl<S: Snapshot> MvccTxn<S> {
                             primary: primary.to_vec(),
                         });
                     }
+                    if should_not_exist {
+                        if write.write_type == WriteType::Put
+                            || (write.write_type != WriteType::Delete
+                                && self.key_exist(&key, write.start_ts - 1)?)
```

Comment:
How about just `if should_not_exist && self.key_exist(&key, ts-1)`?

Reply:
We have got a latest write here, it may don't need to call `key_exist`.